gems/smithy/lib/smithy/model/flattener.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Model
+    # @api private
+    class Flattener
+      def initialize(model)
+        @model = model
+      end
+
+      def shape(id)
+        shape = @model['shapes'][id]
+        return shape unless shape['mixins']
+
+        shape['mixins'].reverse_each do |mixin|
+          mixin_shape = shape(mixin['target'])
+          shape = deep_merge(mixin_shape, shape, exclude_traits(mixin_shape))
+          apply_member_traits(id, shape)
+          shape.delete('mixins')
```

Comment:
Should this be configurable? I could see wanting to resolve the flattened shape, but leave in place mixins (thinking of cases for mixin interface)